src/pretalx/submission/forms/question.py

```diff
@@ -31,8 +33,11 @@ def __init__(self, *args, **kwargs):
         self.queryset = Question.all_objects.filter(event=self.event, active=True)
         if self.target_type:
             self.queryset = self.queryset.filter(target=self.target_type)
+            if self.submission_type:
+                self.queryset = self.queryset.filter(Q(submission_type=self.submission_type) | Q(submission_type=None))
         else:
             self.queryset = self.queryset.exclude(target=QuestionTarget.REVIEWER)
+            self.queryset = self.queryset.filter(Q(submission_type=self.submission_type) | Q(submission_type=None))
```

Comment:
This should only be executed if `self.submission_type` is set and the target_type is submission, since we might not be dealing with a submission question otherwise.